ibacm/src/acm.c

```diff
@@ -2596,17 +2596,64 @@ acm_open_port(struct acmc_port *port, struct acmc_device *dev, uint8_t port_num)
 	port->state = IBV_PORT_DOWN;
 }
 
+static bool dev_support_ib(struct ibv_device *ibdev)
+{
+	struct ibv_context *verbs;
+	struct ibv_device_attr attr;
+	struct ibv_port_attr port_attr;
```

Comment:
Move this decl down to the loop looping through the ports, and use = {} and remove the memset.

But why do you zero-initialize these device and port attrs anyway?

Reply:
Declaration in for loop is C99 syntax. I'd like to put all declarations in the beginning of a function.

zero-initialize those struct to avoid garbage value, which may introduce bugs. It is my coding habit too.